controllers/snapshot/snapshot_adapter.go

```diff
@@ -566,8 +564,18 @@ func (a *Adapter) createCopyOfExistingEnvironment(existingEnvironment *applicati
 
 	err = a.client.Create(a.context, environment)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create environment %s: %w", environment.Name, err)
+		// We don't want to leave the new DTC on the cluster without the matching environment
+		dtcErr := a.client.Delete(a.context, dtc)
+		if dtcErr != nil {
+			a.logger.Error(err, "failed to create ephemeral environment", "environment.name", environment.Name)
```

Comment:
This block puzzles me. It's about DTC deletion failure, but logger is logging failure of env creation. Shouldn't it before `if dtcErr != nil {` ?

Reply:
I did that to prevent both logging the message and returning the error, but I agree it's better to combine the errors into the same message, updated according to the other comment.